src/internal/transactionenv/txncontext/context.go

```diff
@@ -29,6 +30,36 @@ type TransactionContext struct {
 	// PpsJobStopper stops Jobs in any pipelines that are associated with a removed commitset
 	PpsJobStopper  PpsJobStopper
 	PpsJobFinisher PpsJobFinisher
+	FileAccessor   FileAccessor
+}
+
+type identifier interface {
+	WhoAmI(context.Context, *auth.WhoAmIRequest) (*auth.WhoAmIResponse, error)
+}
+
+func New(ctx context.Context, sqlTx *sqlx.Tx, authServer identifier) (*TransactionContext, error) {
+	var username string
+	// check auth once now so that we can refer to it later
+	if authServer != nil {
+		if me, err := authServer.WhoAmI(ctx, &auth.WhoAmIRequest{}); err != nil && !auth.IsErrNotActivated(err) {
+			return nil, err
+		} else if err == nil {
+			username = me.Username
+		}
+	}
```

Comment:
Moving the auth check here might be unexpected for requests that don't ever require auth, but this isn't a problem for user requests, and internal ones should either be given an auth token or maybe using an internal name with the right permissions. It would obviously be easier to just pass in the username string, but I wanted to make it a little harder to accidentally escalate permissions.